controllers/pipeline/pipeline_adapter.go

```diff
@@ -385,8 +436,12 @@ func (a *Adapter) prepareSnapshotForPipelineRun(pipelineRun *tektonv1beta1.Pipel
 	if err != nil {
 		return nil, err
 	}
+	componentSource, err := a.getComponentSourceFromPipelineRun(pipelineRun)
```

Comment:
does this mean, that component doesn't always contain sources, or why there are 2 methods, getting source from component and getting source from pipelinerun?

Reply:
was explained to me offline. Yes we need to fetch data from pipeline, resolving